src/index.ts

```diff
@@ -109,13 +121,21 @@ export function viteSvgToWebfont<T extends GeneratedFontTypes = GeneratedFontTyp
             }
             await generate();
             if (isBuild && !options.inline) {
-                const emitted = processedOptions.types.map<[T, string]>(type => [
-                    type,
-                    `/${this.getFileName(this.emitFile({ type: 'asset', fileName: `assets/${processedOptions.fontName}-${guid()}.${type}`, source: generatedFonts?.[type] }))}`,
-                ]);
+                const emitted = processedOptions.types.map<[T, string]>(type => {
+                    if (!generatedFonts?.[type]) {
+                        throw new Error(`Failed to generate font of type ${type}`);
+                    }
+
+                    const fileContents = Buffer.from(generatedFonts[type]);
+                    const hash = getBufferHash(fileContents);
+                    const filePath = pathJoin(TMP_DIR, `${processedOptions.fontName}-${hash}.${type}`);
+                    ensureDirExistsAndWriteFile(fileContents, filePath); // write font file to a temporary dir
```

Comment:
Here we're writing file to a system temporary dir. So this is workaround for a `this.emitFile()` mechanism which require us to wait a render-hooks which is called too lately.

Because we're emitting the file explicitly, Rollup will successfully find it and will not produce any warnings and will not be switched into a fallback logical branch.